libsolidity/interface/ImportRemapper.h

```diff
@@ -0,0 +1,59 @@
+/*
+	This file is part of solidity.
+
+	solidity is free software: you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	solidity is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License
+	along with solidity.  If not, see <http://www.gnu.org/licenses/>.
+*/
+// SPDX-License-Identifier: GPL-3.0
+#pragma once
+
+#include <optional>
+#include <string>
+#include <vector>
+
+namespace solidity::frontend
+{
+
+// Some helper typedefs to make reading the signatures more self explaining.
+using SourceUnitName = std::string;
```

Comment:
Not very happy about those. The second one is not even used.

Reply:
This was an outcome of the discussion we had via voice. @cameel you want to build on top of this PR, right? Can you then also delete the line below this one in your PR? (`SourceCode typedef` - the code that used it was removed.)

Reply:
The reasoning behind was that we deal a lot with `std::string` but meaning something different. My proposal first was one *could* wrap that into `struct WrappedType { std::string value; };` but I feared that would have been too much.

Reply:
>  Can you then also delete the line below this one in your PR?

Yeah, I can change that.

> OTOH I don't mind, we can move back to plain std::string in @cameel's PR then?

I liked the idea of having distinct types for those but now I'm not so sure about it. There are a bit too many of them and they still do not cover everything (e.g. the remappings still contain plain strings because they're not paths, not source unit names, just... fragments). So I'm fine with it as is but I'm also fine with going back to strings. We can sort this out in the refactor PR.